cmd/pbstatuser/main.go

```diff
@@ -66,6 +67,10 @@ func processMessage(ctx context.Context, message *kafka.GenericMessage) {
 
 	sourceId := asm.SourceID
 
+	// Set source id as logging field
+	logger = ptr.To(logger.With().Str("source_id", sourceId).Logger())
+	ctx := ctxval.WithLogger(origCtx, logger)
```

Comment:
Either this or introducing a temporary variable so I could do `logger = &tempVar`